## Haplotype-resolved T2T Genome Assemblies and Pangenome Graph of Pear

- Author: Qionghou Li (liqionghou@njau.edu.cn)

## Introduction

This repository includes the pipeline used for the research presented in the article _"Haplotype-resolved T2T genome assemblies and pangenome graph of pear reveal diverse patterns of allele-specific expression and genomic basis of fruit quality traits"_. The study focuses on pear genome assemblies and the construction of a pangenome graph, highlighting allele-specific expressions and a broad set of genetic variations.



**Special Note: These codes are for my personal use and may require modifications for your species. Please handle them with caution.**



## Repository Overview

The pipeline encompasses several key processes:

1. **Reads Partition**: Classifying and organizing sequencing reads based on their genomic origin.
2. **Allelle Identification**: Identification allelles from two haplotype genomes.
3. **Pangenome Construction**: Building a pangenome graph based on minigraph-cactus apporach
4. **Graph-based SNP Calling**: Utilizing the pangenome for SNP calling across different pear accessions.



## Data

Genome data (including sequence, gff and pangenome graph generated in manuscript) available at Pear Multiomics Database(https://pearomics.njau.edu.cn/). 



## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details. 



## Citation

Please cite this paper if you find our pipeline or script helpful in your study:

##### Haplotype-resolved T2T genome assemblies and pangenome graph of pear reveal diverse patterns of allele-specific expression and genomic basis of fruit quality traits